spec/system/0_available_kudo_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe 'Manage', type: :system do
+  before do
+    employee = create(:employee)
+    login_as(employee, scope: :employee)
+    driven_by(:selenium_chrome_headless)
+    employee.number_of_available_kudos = 0
+  end
+
+  it '0_available_kudo' do
```

Comment:
Co myślisz o bardziej opisowym nazywaniu testów
`it 'does not show create kudo button, when user has no available kudos' do`

no i warto sprawdzić czy user może zrobić kudosa bez tych punktów bo w sumie teraz testujemy "balans punktów czy się dobrze wyświetla' a nie jest to tak kluczowa kwestia jak możliwość przydzielania kudosów w momencie gdy nie powinieneś móc ich przydzielać. Testy automatyczne piszemy głównie po to że jak będziemy coś zmieniać za jakiś czas to chcemy mieć pewność że kluczowe częśći aplikacji działają tak jak trzeba i nie pozwalają na robienie np zakupów bez pieniędzy, i żeby nie przeklikiwać takich tematów ręcznie, bo byśmy pozapominali posprawdzać połowy takich tematów

recomended reading
https://www.betterspecs.org/

Reply:
no właśnie jakbyś zauważył to właśnie po artykule w betterspecs zmieniłem metodę opisywania na bardziej skrótową. Dosłownie pierwszy artykuł o tym mówi :P

Reply:
pierwszy artykuł mówi o pisaniu dokładnie jaką metodę testujesz w teście - to jest dobre podejście do testów jednostkowych, jednakże tutaj testujemy integracyjne (test systemowy który klika w cały system), tak więc jest wiele metod które może wyłożyć ten test

Dlatego lepiej jest opisać dokładnie co się dzieje w takim teście, żebyśmy odpalając 1000 testów wiedzieli z jego opisu co się dzieje.

Ogólnie w tym zadaniu zaproponował bym przetestować czy napewno employee nie moze dać kudosa jak ma 0 available kudos
zaporponował bym np takie opisanie testu

```
context 'when user has 0 available kudos' do
  it 'doesn't allow to give kudo' do
    # test
  end
end
```
oraz w sumie można by przenieść ten test do `[kudo_create_spec.rb](https://github.com/Valar3/employee-recognition-platform-1650969659/blob/Sprint-2-task-3/spec/system/kudo_create_spec.rb)`, ponieważ tam testujemy czy użytkownik może dać kudo, czy nie może

co myślisz o mojej propozycji?

Reply:
Ok, przyjmuję do wiadomości i postaram się lepiej opisywać testy. a co do Twojej propozycji to sam nie wiem z jednej strony to zawsze jeden mniej test do zrobienia ale z drugiej strony tworzy się jeden duży plik. Wiec zapytam Ciebie które podejście jest mniej 'pamięciożerne' ?

Reply:
To z tego co mi wiadomo nie ma znaczenia czy rozbijesz testy na kilka plików czy zrobisz kilka testów w jednym.

Ja lubie robię jeden plik z testami odnośnie jednego tematu - np jednym tematem jest employee-kudo i tam bym wrzucał wszystkie testy na początku jakieś index testy, potem create, potem edit testy itd - i to wszystko w jakimś pliku typu `employee_kudo_spec.rb` jeżeli testujemy logowanie i rejestracje to pewnie `employee_spec.rb`, dla mnie jest to wygodniejsze bo jak będę coś modyfikował w kudosach dla employee, to w jednym pliku edytuje wszystkie testy i nie muszę skakać po plikach. A co do ilości testów, to moim zdaniem lepiej za dużo niż za mało, generalnie zawsze robię jakiś happy_path, czyli sprawdzam czy nadal najważniejszy element biznesowy zawsze działa, no i często testuje edge casy, czyli jakieś przypadki gdzie zakładam że ktoś będzie chciał "hakować" albo po prostu użytkownicy nieraz nie są zbyt bystrzy i trzeba to też przewidzieć. no i takie testy też nas będą zabezpieczać na przyszłość bardziej